packages/material-ui-lab/src/Masonry/Masonry.js

```diff
@@ -0,0 +1,150 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import {
+  createUnarySpacing,
+  getValue,
+  handleBreakpoints,
+  unstable_resolveBreakpointValues as resolveBreakpointValues,
+} from '@material-ui/system';
+import { deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import { styled, useThemeProps } from '@material-ui/core/styles';
+import { getMasonryUtilityClass } from './masonryClasses';
+import MasonryContext from './MasonryContext';
+
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getMasonryUtilityClass, classes);
+};
+
+export const style = ({ styleProps, theme }) => {
+  let styles = {
+    display: 'grid',
+    gridAutoRows: 0,
+    padding: 0,
+    overflow: 'auto',
+    width: '100%',
+    rowGap: 1,
+    boxSizing: 'border-box',
+  };
+
+  const base = {};
+  Object.keys(theme.breakpoints.values).forEach((breakpoint) => {
+    if (styleProps.spacing[breakpoint] != null) {
+      base[breakpoint] = true;
+    }
+  });
+
+  const spacingValues = resolveBreakpointValues({ values: styleProps.spacing, base });
+  const transformer = createUnarySpacing(theme);
+  const spacingStyleFromPropValue = (propValue) => {
+    return {
+      columnGap: getValue(transformer, propValue),
+    };
+  };
+
+  styles = {
+    ...styles,
+    ...handleBreakpoints({ theme }, spacingValues, spacingStyleFromPropValue),
+  };
+
+  const columnValues = resolveBreakpointValues({ values: styleProps.columns, base });
+  const columnStyleFromPropValue = (propValue) => {
+    return {
+      gridTemplateColumns: `repeat(${propValue}, 1fr)`,
+    };
+  };
+
+  styles = deepmerge(styles, handleBreakpoints({ theme }, columnValues, columnStyleFromPropValue));
```

Comment:
Here we `deepmerge` but earlier we just spread and it's unclear why either pattern was chose.

Reply:
Just spreaing is enough in `MasonryItem`, but here in `Masonry` we need a `deepmerge` because we resolve responsive values twice, once for `spacing` and once for `columns`.

Reply:
I'm just referring to `Masonry` here. On line 64 we use `deepmerge` but on line 54 we just spread.

Reply:
We spread on line 54 beacuse we do not need a `deepmerge`: we haven't resolved any responsive values previously and therefore do not have media queries.

Reply:
But this isn't apparent from the code at all nor does anything guard against future change where we might have media queries.

Reply:
Not unnecessarily using `deepmerge` was Marjia's previous comment. Should I revert?